Data_Science/Advanced animated time series analysis/water_cycle_disruption.py

```
@@ -0,0 +1,208 @@
+        elif (frame >= len(data) * 20) and frame < len(data) * 40:
+            lines[param].set_data(interpolated_data['Year'][:frame + 1 - len(data) * 20], interpolated_data[param][:frame + 1 - len(data) * 20])
+        elif frame >= len(data) * 40:
+            reg_lines[param].set_data(future_years[:frame+1-len(data) * 40], regression_lines[param][:frame+1-len(data) * 40])
+
+    return list(scatters.values()) + list(lines.values()) + list(reg_lines.values())
+
+scatters = {}
+lines = {}
+reg_lines = {}
+
+from matplotlib.animation import FuncAnimation, PillowWriter
```

Ruff (0.11.9)

Redefinition of unused `FuncAnimation` from line 5

Remove definition: `FuncAnimation`

(F811)